### PR TITLE
[Term Entry] Dart Iterable: New Concept Entry Iterable.md

### LINKED ISSUE #4245: [Entry] Iterable: New concept entry `iterable.md`
### Reason/inspiration (optional)

We would like a new term entry on the `Iterable` term for [Dart](https://www.codecademy.com/resources/docs/dart). The entry should go under the following path `docs/content/dart/concepts/iterable/iterable.md`.

The entry should describe the term and how it is used within Dart. Also include: 
- Add a "Syntax" section.
- An "Example" section that explains iterables in Dart.
- Explain how iterables are different from lists in Dart.

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by AlessioAltomonte**
### Description

- Added the Iterable term file for Dart 
- Closes #4245 
- 
### Type of Change

- Adding a new entry

### Checklist

- [x ] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x ] My changes generate no new warnings.
- [x ] I have performed a self-review of my own writing and code.
- [x ] I have checked my entry and corrected any misspellings.
- [x ] I have made corresponding changes to the documentation if needed.
- [x ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x ] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/dart/concepts/iterable/iterable.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+Each time you access the iterator getter, the iterable object returns a new iterator. For this reason, you can create more than one iterator from the same iterable object.
+
+If an object is iterable, you can use the for-in statement to iterate over its elements.
+
+## Syntax
+
+[Text, code, images, parameters, etc. about the syntax]
+
+## Example
```
Comment (Sriparno08) [inline content/dart/concepts/iterable/iterable.md:28]: Let's remove the 'Syntax'  section as it's not needed:

```suggestion
## Example
```

File: content/dart/concepts/iterable/iterable.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+Code language: Dart (dart)
+
+Output:
+
+2
+
+## Summary
+
+    An iterable is a collection of elements that can be accessed sequentially.
+    Dart uses the Iterable<E> abstract class to represent iterable objects.
```
Comment (Sriparno08) [inline content/dart/concepts/iterable/iterable.md:66]: Let's limit this entry to a single example as it's enough for demonstrating the concept. Plus, let's remove the 'Summary' section as it's not needed.

Also, let's replace the whole with the following:

````suggestion
Here is the output for the above code:

```shell
12
23
34
45
56
```
````

File: content/dart/concepts/iterable/iterable.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+Since List and Set are iterable, you can use the for-in statement to iterate over their elements. For example:
+
+void main() {
+Iterable<int> iterable = [1, 2, 3, 4, 5];
+for (var n in iterable) {
+print(n);
+}
+}
+Code language: Dart (dart)
```
Comment (Sriparno08) [inline content/dart/concepts/iterable/iterable.md:38]:
````suggestion
```dart
void main() {
  Iterable<int> iter = [12, 23, 34, 45, 56];
  for (var num in iter) {
    print(num);
  }
}
```
````

**Review by Sriparno08 — COMMENTED** [review]
Hey @AlessioAltomonte, thanks for contributing to Codecademy!

I've listed all the modifications that you need to make in your entry below:

**Comment by Sriparno08** [comment]
Hey @AlessioAltomonte, please make the required changes so that we can review them and move on to the second round of review.

**Review by Sriparno08 — APPROVED** [review]
This looks good for a second review 👍

File: content/dart/concepts/iterable/iterable.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  - 'Objects'
+  - 'Loops'
+CatalogContent:
+  - 'learn-dart'
+  - 'paths/computer-science'
+---
+
+An **iterable** is a group of elements which can be retrieved sequentially. Dart makes use of the `Iterable<E>` abstract class for representing iterable objects. The elements inside these objects can be traversed using the `for-in` loop.
+
+An iterable object consists of an iterator getter, which returns an iterator object. The returned iterator is then used by the iterable object for element traversal. Each time the iterator getter is accessed, a new iterator is returned by the iterable object. This means that more than one iterator can be returned by the same iterable object.
```
Comment (ishg-153) [inline content/dart/concepts/iterable/iterable.md:18]:
```suggestion
An iterable object consists of an iterator getter, which returns elements from the iterable object. The iterator is used for element traversal. Each time the iterator getter is accessed, a new iterator is returned by the iterable object. This means that more than one iterator can be returned by the same iterable object.
```

**Review by ishg-153 — CHANGES_REQUESTED** [review]
@AlessioAltomonte I have suggested changes please make those. Let me know once done.

**Comment by ishg-153** [comment]
@AlessioAltomonte It's been 2 weeks. Are you still working on the changes?

**Review by ishg-153 — APPROVED** [review]
Went ahead and made the changes.
This is ready to merge.

**Pull request merged**